core/src/main/java/io/undertow/server/handlers/resource/PathResourceManager.java

```diff
@@ -194,7 +194,7 @@ public Resource getResource(final String p) {
             path = p;
         }
         try {
-            Path file = fileSystem.getPath(base, path);
+            Path file = fileSystem.getPath(base, path).normalize();
             String normalizedFile = file.normalize().toString();
```

Comment:
There should only be one call to normalize(), but in general we need to be really careful here as there are security issues around symlinks. In your failing example is there a symlink somewhere on the path?

Reply:
My failing does not have any symlinks on the path.  Quite simply the resource gets requested with a false directory.  It has to do with the inner workings of struts 1.

Any thoughts as to what's changed between the undertow subsystem in JBoss EAP 7.2 and the old subsystem in JBoss EAP 6.2?  Although I would agree that the pathing is dumb, it's in a very large scale application that would require hundreds of changes (or ~70 empty directories created at the root of the WAR file) in order to work using the new system based on my tests.

Any/All help is greatly appreciated :-)